var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -274,25 +294,35 @@ class Llvm(CMakePackage, CudaPackage):
     # Internal compiler error on gcc 8.4 on aarch64 https://bugzilla.redhat.com/show_bug.cgi?id=1958295
     conflicts("%gcc@8.4:8.4.9", when="@12: target=aarch64:")
 
-    # When these versions are concretized, but not explicitly with +libcxx, these
-    # conflicts will enable clingo to set ~libcxx, making the build successful:
-
-    # libc++ of LLVM13, see https://libcxx.llvm.org/#platform-and-compiler-support
-    # @13 does not support %gcc@:10 https://bugs.llvm.org/show_bug.cgi?id=51359#c1
-    # GCC    11     - latest stable release per GCC release page
-    # Clang: 11, 12 - latest two stable releases per LLVM release page
-    # AppleClang 12 - latest stable release per Xcode release page
-    conflicts("%gcc@:10", when="@13:+libcxx")
-    conflicts("%clang@:10", when="@13:+libcxx")
-    conflicts("%apple-clang@:11", when="@13:+libcxx")
-
-    # libcxx-4 and compiler-rt-4 fail to build with "newer" clang and gcc versions:
-    conflicts("%gcc@7:", when="@:4+libcxx")
-    conflicts("%clang@6:", when="@:4+libcxx")
-    conflicts("%apple-clang@6:", when="@:4+libcxx")
-    conflicts("%gcc@7:", when="@:4+compiler-rt")
-    conflicts("%clang@6:", when="@:4+compiler-rt")
-    conflicts("%apple-clang@6:", when="@:4+compiler-rt")
+    # libcxx=project imposes compiler conflicts
+    # see https://libcxx.llvm.org/#platform-and-compiler-support for the latest release
+    # and https://github.com/llvm/www-releases for older releases
+    with when("libcxx=project"):
```

Comment:
@tgamblin we need a way to deal with useful things like this for the caching stuff, all the inputs are literal, but the directive calls receive variables.  Maybe something we could explore https://pypi.org/project/pystarlark/ for, make an invocation of pyskylark with a literal containing directives behave like invocations of those literals? Or, easier, make it so we can import directives from a flat yaml or json value or better dhall or cue document or something?  We need to be able to do this stuff, but not being able to cache llvm (given its ubiquity and complexity) would be really unfortunate. =/